src/wp-includes/admin-bar.php

```diff
@@ -1224,33 +1224,54 @@ function wp_admin_bar_add_secondary_groups( $wp_admin_bar ) {
 	);
 }
 
+
+
 /**
- * Prints style and scripts for the admin bar.
+ * Enqueues inline style to hide the admin bar when printing.
  *
- * @since 3.1.0
+ * @since 6.4.0
  */
-function wp_admin_bar_header() {
-	$type_attr = current_theme_supports( 'html5', 'style' ) ? '' : ' type="text/css"';
-	?>
-<style<?php echo $type_attr; ?> media="print">#wpadminbar { display:none; }</style>
-	<?php
+function wp_enqueue_admin_bar_header_styles() {
+	// Back-compat for plugins that disable functionality by unhooking this action.
+	$action = is_admin() ? 'admin_head' : 'wp_head';
+	if ( ! has_action( $action, 'wp_admin_bar_header' ) ) {
+		return;
+	}
+	remove_action( $action, 'wp_admin_bar_header' );
+
+	wp_add_inline_style( 'admin-bar', /* language=CSS */ '@media print { #wpadminbar { display:none; } }' );
 }
 
 /**
- * Prints default admin bar callback.
+ * Enqueues inline bump styles to make room for the admin bar.
  *
- * @since 3.1.0
+ * @since 6.4.0
  */
-function _admin_bar_bump_cb() {
-	$type_attr = current_theme_supports( 'html5', 'style' ) ? '' : ' type="text/css"';
-	?>
-<style<?php echo $type_attr; ?> media="screen">
-	html { margin-top: 32px !important; }
-	@media screen and ( max-width: 782px ) {
-		html { margin-top: 46px !important; }
+function wp_enqueue_admin_bar_bump_styles() {
+	if ( current_theme_supports( 'admin-bar' ) ) {
+		$admin_bar_args  = get_theme_support( 'admin-bar' );
+		$header_callback = $admin_bar_args[0]['callback'];
+	}
+
+	if ( empty( $header_callback ) ) {
+		$header_callback = '_admin_bar_bump_cb';
+	}
+
+	if ( '_admin_bar_bump_cb' !== $header_callback ) {
+		return;
 	}
```

Comment:
Just noting that this also accounts for this special case here, when the `$header_callback` is `__return_false`: 

https://github.com/WordPress/wordpress-develop/blob/fa0969d06a2e0178c780143a5ac2ec7ea03bb216/src/wp-includes/class-wp-admin-bar.php#L57-L64

See themes: https://wpdirectory.net/search/01HAWFD8X1MB2HKBXKHNHZB14J

When that case happens, this function will continue to short-circuit as it should.